### PR TITLE
RD-5782 Migrated sites widget

### PR DESCRIPTION
**Pull request by Vorbert-Kruk**
## Description

Changes consists of:
- Migrating sites widget to typescript
- Migrating away from implicit `lodash` imports
- Migrated away from using acronyms as a variable names 🎸 

Labels migration to the translation file will take place in a separate PR (#2434) 💪 

## Screenshots / Videos

N/A

## Checklist

- [x] My code follows the [UI Code Style](https://cloudifysource.atlassian.net/l/c/x8fq902p).
- [x] I followed the [UI Testing Guidelines](https://cloudifysource.atlassian.net/l/cp/udCVfvrx) and verified that all tests/checks have passed.
- [x] I verified if that change requires any update in the [documentation](https://cloudifysource.atlassian.net/l/c/4XKtPocy).
- [x] I added proper labels to this PR.

## Tests

 

[Build Status](https://jenkins.cloudify.co/view/UI%20Health%20View/job/Stage-UI-System-Test/4509/)

## Documentation

N/A

### CODE REVIEW

File: widgets/sites/src/widget.tsx [hunk 1]
```diff
@@ -1,8 +1,11 @@
-// @ts-nocheck File not migrated fully to TS
+import { isEmpty, get } from 'lodash';
 import SitesTable from './SitesTable';
 import './widget.css';
+import type { SitesWidget } from './widgets.types';
 
-Stage.defineWidget({
+// TODO Norbert: Migrate labels to translation file - probably as a separate PR
```
Comment (Vorbert-Kruk) [inline widgets/sites/src/widget.tsx:6]: ### Note
This will be completed in a separate PR 🎉

File: widgets/sites/src/widget.tsx [hunk 2]
```diff
@@ -23,26 +26,31 @@ Stage.defineWidget({
         Stage.GenericConfig.SORT_ASCENDING_CONFIG(true)
     ],
 
-    render(widget, data, error, toolbox) {
+    render(widget, data, _error, toolbox) {
         const { Loading } = Stage.Basic;
 
-        if (_.isEmpty(data)) {
+        if (!data || isEmpty(data)) {
```
Comment (qooban) [inline widgets/sites/src/widget.tsx:32]: Maybe:
```suggestion
        if (Stage.Utils.isEmptyWidgetData(data)) {
```
?

File: widgets/sites/src/SiteLocationInput.tsx [hunk 3]
```diff
@@ -1,17 +1,23 @@
...
     const [enteredValue, setEnteredValue] = useState(value || '');
 
     useEffect(() => setEnteredValue(value), [value]);
 
-    function onLocationChange(newValue) {
-        setEnteredValue(newValue);
-        onChange(null, { name: 'siteLocation', value: newValue });
+    function onLocationChange(changedLocation: string) {
+        setEnteredValue(changedLocation);
+        onChange(null, { name: 'siteLocation', value: changedLocation });
```
Comment (qooban) [inline widgets/sites/src/SiteLocationInput.tsx:20]: Optional:

I would go with `newValue` or `newLocation` as it seems more common (than `changed*`) pattern we try to follow in this repo:

```suggestion
    function onLocationChange(newLocation: string) {
        setEnteredValue(newLocation);
        onChange(null, { name: 'siteLocation', value: newLocation });
```

File: widgets/sites/src/SiteActions.ts [hunk 4]
```diff
@@ -1,23 +1,32 @@
...
-    constructor(toolbox) {
+    toolbox: Stage.Types.WidgetlessToolbox;
+
+    constructor(toolbox: Stage.Types.WidgetlessToolbox) {
         this.toolbox = toolbox;
     }
 
-    doGet(name) {
+    doGet(name: string) {
         return this.toolbox.getManager().doGet(`/sites/${name}`);
```
Comment (qooban) [inline widgets/sites/src/SiteActions.ts:12]: With something like (haven't tested, don't know exact response)
```suggestion
        return this.toolbox.getManager().doGet<Site>(`/sites/${name}`);
```
you'll be able to easier type places which uses this method.

File: widgets/sites/src/SiteLocationMap.tsx [hunk 5]
```diff
@@ -66,25 +96,11 @@ class SiteLocationMap extends React.Component {
                 zoom={initialZoom}
                 center={toLatLng(this.initialLocation)}
             >
-                <Leaflet.TileLayer attribution={attribution} url={url} />
```
Comment (qooban) [inline widgets/sites/src/SiteLocationMap.tsx:69]: Hm... Are you sure about removing that prop? 🤔 
We have `attribution` property in user configuration for maps.

File: widgets/sites/src/SiteActions.ts (hunk 4 above)
Comment (Vorbert-Kruk) [reply widgets/sites/src/SiteActions.ts:12]: As I'm investigating right now, this method is not being used, so I'll just remove it 😁

File: widgets/sites/src/widget.tsx [hunk 6]
```diff
@@ -23,26 +26,31 @@ Stage.defineWidget({
...
-            items: _.map(sites.items, site => {
+            items: sites.items.map(site => {
                 return {
                     ...site,
                     created_at: Stage.Utils.Time.formatTimestamp(site.created_at),
                     deploymentCount: deploymentsPerSite[site.name] || 0
                 };
             }),
-            total: _.get(sites, 'metadata.pagination.total', 0)
+            total: get(sites, 'metadata.pagination.total', 0)
```
[the remark this answers is not in the document]
Comment (Vorbert-Kruk) [reply widgets/sites/src/widget.tsx:53]: I've checked it and it seems like a correct assumption 💪 
I'll adjust it 🎉

File: widgets/sites/src/widget.tsx (hunk 2 above)
Comment (Vorbert-Kruk) [reply widgets/sites/src/widget.tsx:32]: Woah, I didn't know about this one, looks awesome! 🤩 
Definitely will implement 🎉

File: widgets/sites/src/SiteLocationInput.tsx (hunk 3 above)
Comment (Vorbert-Kruk) [reply widgets/sites/src/SiteLocationInput.tsx:20]: Sure, if that's more common approach, then let's go with it 💪 
Thanks btw for propagating that knowledge 🎉

File: widgets/sites/src/SiteLocationMap.tsx (hunk 5 above)
Comment (Vorbert-Kruk) [reply widgets/sites/src/SiteLocationMap.tsx:69]: Oh, I thought that this prop is not being used
`store` line tricked me a little bit - thanks for pointing that out and for the explanation about the use case of that piece of code 🎉

File: widgets/sites/src/widgets.types.ts [hunk 7]
```diff
@@ -0,0 +1,39 @@
+import type { Visibility } from 'app/widgets/common/types';
+import type { DataTableConfiguration, PollingTimeConfiguration } from '../../../app/utils/GenericConfig';
+
+export interface FetchedSite {
+    /* eslint-disable camelcase */
+    created_at: Date;
```
[the remark this answers is not in the document]
Comment (Vorbert-Kruk) [reply widgets/sites/src/widgets.types.ts]: Oh old habbits could've taken over 😄 
In previous projects we were treating and typing dates (retrieved from backends API) as `Date`
If I remember properly, we had some middlewares for that 🤔 

But nonetheless, this kind of "no-brain typing" took over this time 😅 
You're right, those kind of properties should be treated as `string`
I'll adjust it right away 💪

[the remark this answers is not in the document]
Comment (Vorbert-Kruk) [reply widgets/sites/src/widgets.types.ts]: Addressed as a part of 77747fae3d5f598156d005d1baedf270698aad8f

File: widgets/sites/src/widget.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (Vorbert-Kruk) [reply widgets/sites/src/widget.tsx]: Addressed as a part of 77747fae3d5f598156d005d1baedf270698aad8f

File: widgets/sites/src/SiteLocationMap.tsx (hunk 5 above)
Comment (Vorbert-Kruk) [reply widgets/sites/src/SiteLocationMap.tsx]: Addressed as a part of 77747fae3d5f598156d005d1baedf270698aad8f

**Review by qooban — APPROVED** [review]
